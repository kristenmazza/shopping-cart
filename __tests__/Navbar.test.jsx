import React from "react";
import Navbar from "../src/components/Navbar";
import { describe, it, expect } from 'vitest';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter } from "react-router-dom";

describe("Navbar component", () => {
    it("tests the children inside Navbar component", () => {
        const navbar = TestRenderer.create(<MemoryRouter><Navbar /></MemoryRouter>).toJSON();

        // Shop route
        expect(navbar.children[1].children[0].children[0].children[0].type).toBe('a');
        expect(navbar.children[1].children[0].children[0].children[0].props.href).toBe('/');

        // About route
        expect(navbar.children[1].children[0].children[1].children[0].type).toBe('a');
        expect(navbar.children[1].children[0].children[1].children[0].props.href).toBe('/shop');
    });

    it("will create a snapshot", () => {
        const navbar = TestRenderer.create(<MemoryRouter><Navbar /></MemoryRouter>).toJSON();
        expect(navbar).toMatchSnapshot();
    })
});
