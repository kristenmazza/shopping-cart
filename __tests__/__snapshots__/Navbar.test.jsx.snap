// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > will create a snapshot 1`] = `
<nav
  className="_navigation_39d119"
>
  <button
    className="_hamburger_39d119"
    onClick={[Function]}
  >
    <svg
      className="h-5 w-5"
      fill="white"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM9 15a1 1 0 011-1h6a1 1 0 110 2h-6a1 1 0 01-1-1z"
        fillRule="evenodd"
      />
    </svg>
  </button>
  <div
    className="_navigationMenu_39d119 "
  >
    <ul>
      <li>
        <a
          href="/shop"
          onClick={[Function]}
        >
          Shop
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
